src/io/internal/pipelines/dicom/apply-pstate-to-image.cxx

```diff
@@ -0,0 +1,582 @@
+          case DVPSL_polylineCurve: type = "polylineCurve"; break;
+        }
+        curveJson.AddMember("Type", Value(type.c_str(), a), a);
+        curveJson.AddMember("AxisUnitsX", Value(pcurve->getCurveAxisUnitsX(), a), a);
+        curveJson.AddMember("AxisUnitsY", Value(pcurve->getCurveAxisUnitsY(), a), a);
+        curveJson.AddMember("Label", Value(StringRef(pcurve->getCurveLabel())), a);
+        curveJson.AddMember("Description", Value(StringRef(pcurve->getCurveDescription())), a);
+
+        j = pcurve->getNumberOfPoints();
+        double dx=0.0, dy=0.0;
+        std::vector<double> points;
+        points.reserve(j*2);
+        for (i=0; i<j; i++)
+        {
+          if (EC_Normal==pcurve->getPoint(i,dx,dy))
+          {
+            points.push_back(dx);
+            points.push_back(dy);
+          }
+        }
+        curveJson.AddMember("Points", getArrayJson(points, a), a);
+        curvesJsonArray.PushBack(curveJson, a);
+      }
+    }
+
+    layerJson.AddMember("Curves", curvesJsonArray, a);
+
+    // overlay objects
+    const void *overlayData=NULL;
+    unsigned int overlayWidth=0, overlayHeight=0, overlayLeft=0, overlayTop=0;
+    OFBool overlayROI=OFFalse;
+    Uint16 overlayTransp=0;
+    char overlayfile[100];
+    FILE *ofile=NULL;
+
+    max = ps.getNumberOfActiveOverlays(layer);
+
+    // Overlays[]
+    Value overlaysJsonArray(kArrayType);
+
+    for (size_t ovlidx=0; ovlidx<max; ovlidx++)
+    {
+      Value overlayJson(kObjectType);
+      overlayJson.AddMember("Index", Value(static_cast<uint32_t>(ovlidx)), a);
+      std::stringstream value;
+      value << "0x" << STD_NAMESPACE hex << ps.getActiveOverlayGroup(layer, ovlidx) << STD_NAMESPACE dec;
+      overlayJson.AddMember("Group", Value(value.str().c_str(), a), a);
+      overlayJson.AddMember("Label", Value(StringRef(ps.getActiveOverlayLabel(layer, ovlidx))), a);
+      overlayJson.AddMember("Description", Value(StringRef(ps.getActiveOverlayDescription(layer, ovlidx))), a);
+      overlayJson.AddMember("Type", Value(StringRef(ps.activeOverlayIsROI(layer, ovlidx) ? "ROI" : "graphic")), a);
+
+      /* get overlay data */
+      if (EC_Normal == ps.getOverlayData(layer, ovlidx, overlayData, overlayWidth, overlayHeight,
+          overlayLeft, overlayTop, overlayROI, overlayTransp))
+      {
+        overlayJson.AddMember("Width", Value(overlayWidth), a);
+        overlayJson.AddMember("Height", Value(overlayHeight), a);
+        overlayJson.AddMember("Left", Value(overlayLeft), a);
+        overlayJson.AddMember("Top", Value(overlayTop), a);
+
+        std::stringstream buff;
+        buff << "P5\n" << overlayWidth << " " << overlayHeight << " " << "255\n";
+        buff.write((const char*)overlayData, overlayWidth * overlayHeight);
+
+        constexpr bool urlFriendly = false;
+        overlayJson.AddMember("OverlayData", Value(base64_encode(buff.str(), urlFriendly).c_str(), a), a);
+      } else {
```

Comment:
todo check: width, height?

Reply:
width, height properties are set in the same `overlayJson` object separately.